controllers/help_request.js

```diff
@@ -5,6 +5,45 @@ var UserModel = require("../models/user").model;
 var NotificationService = require("../services/notification.service");
 var UserService = require("../services/user.service");
 
+const putHelpRequest = async (req, res) => {
+  let status = req.body.status;
+  let newResponder = req.body.newResponder;
+  let helpReqId = req.body.helpReqId;
+
+  if (!(status=="open"||status=="sent_to_responder"||status=="taken"||status=="arrived"||status=="resolved")){
+    handle.badRequest(res, "Incorrect status")
+  }
+  else{
+    try {
+      var help_request = await HelpRequestModel.findOne({
+        _id: new ObjectId(helpReqId)
+      })
+    } catch (err) {
+      handle.badRequest(res, err.message);
+    }
+
+    if (help_request==null)
+      handle.badRequest(res,"Help Request does not exist")
+    else{
+      help_request.responderIds.push({_id:false, id: newResponder});
```

Comment:
Please check if the id exists in the array first before pushing to it

Reply:
Changed appropriately